File: app/src/main/java/expressive/fan/states/PlayingState.java
package expressive.fan.states;

import expressive.fan.core.MainActivityController;

public class PlayingState extends AbstractState {

    private MainActivityController controller;

    public PlayingState(MainActivityController controller) {
        this.controller = controller;
        this.controller.setPlayButtonEnabled(false);
        this.controller.setPauseButtonEnabled(true);
        this.controller.setStopButtonEnabled(true);
    }

    @Override
    public void pauseClicked() {
        controller.pause();
        controller.setNextState(new PausedState(controller));
    }

    @Override
    public void stopClicked() {
        controller.stop();
        controller.setNextState(new StoppedState(controller));
    }

    @Override
    public void progressChanged(int progress) {
        int total = controller.getTotalAudioDuration();
        int newPos = total / 100 * progress;
        controller.seekTo(newPos);
    }
}

File: app/src/main/java/expressive/fan/states/PlayingState.kt
package expressive.fan.states

import expressive.fan.core.MainActivityController

class PlayingState(private val controller: MainActivityController) : AbstractState() {
    override fun pauseClicked() {
        controller.pause()
        controller.setNextState(PausedState(controller))
    }

    override fun stopClicked() {
        controller.stop()
        controller.setNextState(StoppedState(controller))
    }

    override fun progressChangedByUser(progress: Int) {
        controller.seekToNewPos(progress)
    }

    override fun progressChangedByScheduler() {
        controller.updateProgress()
    }

    override fun completed() {
        controller.setNextState(StoppedState(controller))
        controller.stop()
    }

    init {
        controller.setPlayButtonEnabled(false)
        controller.setPauseButtonEnabled(true)
        controller.setStopButtonEnabled(true)
    }
}